src/apis/fetchCategories.ts

```diff
@@ -0,0 +1,43 @@
+import { Category } from "../types";
+
+export const fetchCategories = async (): Promise<Category[]> => {
+    return await fetch("http://localhost:3001/categories")
+        .then((response) => response.json())
+};
+
+export const fetchDeleteCategory = async (id: string) => {
+    await fetch(`http://localhost:3001/categories/${id}`, {
```

Comment:
better to return the fetch  instead of async/ await.
this is arguable since it returns nothing, but in case of failure / throwing an exception the callee should know about that error to handle it. but when ever we have returned response its better to return 
and the caller  can use either await or promise chaining to handle the result.
``` return fetch ```